src/Maestro/tests/Maestro.ScenarioTests/EndToEndFlowLogic.cs

```diff
@@ -0,0 +1,352 @@
+                                await GitCommitAsync("Add dependencies");
+                                await using (await PushGitBranchAsync("origin", targetBranch))
+                                {
+                                    TestContext.WriteLine("Trigger the dependency update");
+                                    await TriggerSubscriptionAsync(subscription1Id.Value);
+                                    await TriggerSubscriptionAsync(subscription2Id.Value);
+
+                                    List<DependencyDetail> expectedDependencies = expectedDependenciesSource1.Concat(expectedDependenciesSource2).ToList();
+
+                                    TestContext.WriteLine($"Waiting on PR to be opened in {targetRepoUri}");
+
+                                    if (isAzDoTest)
+                                    {
+                                        await CheckBatchedAzDoPullRequest(source1RepoName, source2RepoName, targetRepoName, targetBranch, expectedDependencies, reposFolder.Directory);
+                                    }
+                                    else
+                                    {
+                                        await CheckBatchedGitHubPullRequest(targetBranch, source1RepoName, source2RepoName, targetRepoName, expectedDependencies, reposFolder.Directory);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public async Task NonBatchedFlowTestBase(string targetBranch, string channelName, bool isAzDoTest, bool allChecks = false, bool isCoherencyTest = false)
+        {
+            string targetRepoName = testRepo2Name;
+            string sourceRepoName = testRepo1Name;
+            string childRepoName = testRepo2Name;
+
+            if (isCoherencyTest)
+            {
+                targetRepoName = testRepo3Name;
+                sourceBranch = "coherency-tree";
+                sourceCommit = "cc1a27107a1f4c4bc5e2f796c5ef346f60abb404";
+                source2Commit = "8460158878d4b7568f55d27960d4453877523ea6";
+            }
+
+            string testChannelName = channelName;
+            string sourceRepoUri = isAzDoTest ? GetAzDoRepoUrl(sourceRepoName) : GetRepoUrl(sourceRepoName);
+            string targetRepoUri = isAzDoTest ? GetAzDoRepoUrl(targetRepoName) : GetRepoUrl(targetRepoName);
+            string childRepoUri = GetRepoUrl(childRepoName);
+
+            TestContext.WriteLine($"Creating test channel {testChannelName}");
+            await using (AsyncDisposableValue<string> testChannel = await CreateTestChannelAsync(testChannelName).ConfigureAwait(false))
+            {
+                await using (AsyncDisposableValue<string> subscription1Id = allChecks ? await CreateSubscriptionAsync(testChannelName, testRepo1Name, testRepo2Name, targetBranch,
```

Comment:
This line probably needs to be broken up into a few different lines, it's pretty darn complicated.

Reply:
Do you mean broken up visually or split into an if/else for clarity?

Reply:
Split into an if/else for clarity. A using that's got a ternary that's got some awaits... that's a lot to take in. :-)  But if visially it could get broken up a bit maybe that would be good enough.